src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -261,22 +168,181 @@ contract LandBaseTokenV3 is ERC721BaseTokenV2 {
+    }
+
+    /// @notice x coordinate of Land token
+    /// @param id tokenId
+    /// @return the x coordinates
+    function getX(uint256 id) external pure returns (uint256) {
+        return _getX(id);
+    }
+
+    /// @notice y coordinate of Land token
+    /// @param id tokenId
+    /// @return the y coordinates
+    function getY(uint256 id) external pure returns (uint256) {
+        return _getY(id);
+    }
+
+    /// @notice check whether address `who` is given minter rights.
+    /// @param who The address to query.
+    /// @return whether the address has minter rights.
+    function isMinter(address who) public view returns (bool) {
+        return _minters[who];
+    }
+
+
+    /// @notice checks if Land has been minted or not
+    /// @param size size of the
+    /// @param x x coordinate of the quad
+    /// @param y y coordinate of the quad
+    /// @return bool for if Land has been minted or not
+    function exists(
+        uint256 size,
+        uint256 x,
+        uint256 y
+    ) public view returns (bool) {
+        require(x % size == 0 && y % size == 0, "Invalid coordinates");
+        require(x <= GRID_SIZE - size && y <= GRID_SIZE - size, "Out of bounds");
+        return _ownerOfQuad(size, x, y) != address(0);
+    }
+
+    function _mintAndTransferQuad(
+        address to,
+        uint256 size,
+        uint256 x,
+        uint256 y,
+        bytes memory data
+    ) internal {
+        require(to != address(0), "to is zero address");
+        require(isMinter(msg.sender), "Only a minter can mint");
+
+        (uint256 layer, , ) = _getQuadLayer(size);
+        uint256 quadId = _getQuadId(layer, x, y);
+
+        // Length of array is equal to number of 3x3 child quad a 24x24 quad can have
+        Land[] memory quadMinted = new Land[](64);
+        uint256 index;
+        uint256 landMinted;
+
+        if(size > 3)(index,landMinted) = _checkAndClearOwner(Land({x: x, y: y, size: size}), quadMinted, landMinted, index, size/2);
+
+        for (uint256 i = 0; i < size * size; i++) {
+            uint256 _id = _idInPath(i, size, x, y);
+            uint256 xi = _getX(_id);
+            uint256 yi = _getY(_id);
+            bool isAlreadyMinted = _isQuadMinted(quadMinted, xi, yi, 1, index);
+            if (isAlreadyMinted) {
+                emit Transfer(msg.sender, to, _id);
+            } else {
+                if (_owners[_id] == uint256(msg.sender)) {
+                    _owners[_id] = 0;
+                    landMinted += 1;
+                    emit Transfer(msg.sender, to, _id);
+                } else {
+                    require(_owners[_id] == 0, "Already minted");
+                    emit Transfer(address(0), to, _id);
+                }
+            }
+        }
+
+        _owners[quadId] = uint256(to);
+        _numNFTPerAddress[to] += size * size;
+        _numNFTPerAddress[msg.sender] -= landMinted;
+
+        _checkBatchReceiverAcceptQuad(msg.sender, address(0), to, size, x, y, data);
```

Comment:
The call to _checkBatchReceiverAcceptQuad with address(0) is ok for minted and transferred lands, or it must be something different depending the case ?

Reply:
Yes, this seems alright to me, as this is also being used in functions which only do mint and also in functions which only do transfers.

Reply:
The problem I see is that in transferQuad the call is made: `_checkBatchReceiverAcceptQuad(_msgSender(), from, to, size, x, y, data);`  and from is not address(0) for the transferred lands

Reply:
This is checked for the final quad being minted, though it would have child quad already minted the final parent quad would be minted here. so from is address(0).

Reply:
It is very complicated, right ?
You have:
-  transferred quads: for those you know the size. The call must be done inside _checkAndClearOwner 
- minted lands that must be described as quads (if we want to use _checkBatchReceiverAcceptQuad)